### DISCUSSION TITLE
Adjust animation-duration is currently not possible

### DISCUSSION THREAD

**Discussion by jonasdev** [tailwindlabs/tailwindcss#9730] [Ideas]
I am using `animation-pulse` for an icon. Looks great.
But I want to adjust the `animation-duration` to be a tiny bit slower.
It is `2000ms` by default, I would like it to be `3000ms`

Currently I've tried with `duration-[3000ms]` - which doesn't have any effect, since this is `transform-duration` and **not** `animation-duration`.

I think the class for `animation-duration` is missing?
Something like `animate-pulse-[3000ms]`

**Comment by kg-currenxie** [comment]
Would love this too! `animate-spin` is too slow for us :D

**Reply by jonasdev** [reply]
One thing is speed adjustment. But I also wonder why `duration-[3000ms]` is `transform-duration` over `animation-duration` 

Wouldn't `t-duration-[3000ms]` and `a-duration-[3000ms]` be a more tailwind way of doing it?

**Reply by maxisix** [reply]
Do you guys know if it will be included soon?

**Reply by jonasdev** [reply]
Doesn't seem like it. No one from the tailwind core has commented, so guess not :)

**Reply by 0-Chan** [reply]
Still no solution? I think that's a fair question.

**Reply by renanrcrr** [reply]
To adjust the animation duration for the animation-pulse in Tailwind CSS, you'll need to customize it using Tailwind's configuration. Tailwind doesn't have a built-in utility for directly setting the animation-duration, so you'll need to extend the default configuration to add your custom duration.

**Reply by jonasdev** [reply]
Thats my point. It is like it is missing, since it is made for transform-duration but not animation-duration.

**Reply by mazei513** [reply]
We worked around this with arbitrary properties for now, e.g. `"animate-spin [animation-duration:300ms]"`.

**Reply by isaacfennell** [reply]
Seems like an easy addition, aside from the naming convention issue above

**Reply by dicash** [reply]
We moved to v4 with --animate-* variables, and this property would be a game changer

**Reply by sir-captainmorgan21** [reply]
We just created utilities. Its like 4 of them. Should be super easy for them to add, and no naming conflicts. Could do the same for animation duration, though obviously need to support a variable value
```
@utility fill-mode-none {
  animation-fill-mode: none;
}

@utility fill-mode-forwards {
  animation-fill-mode: forwards;
}

@utility fill-mode-backwards {
  animation-fill-mode: backwards;
}

@utility fill-mode-both {
  animation-fill-mode: both;
}
```

**Comment by jkiss** [comment]
and animation duration can NOT custom in theme block
